src/ws2st/refactor.hpp

```diff
@@ -0,0 +1,243 @@
+/**
+ * @file refactor.hpp
+ * @author james
+ * @brief Idea is to write out the results of the refactoring using the classes defined here
+ * @version 0.1
+ * @date 2024-01-22
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#pragma once
+
+#include "common.hpp"
+
+namespace wash {
+
+    std::optional<std::string> getSourceText(ASTContext* ctx, SourceRange srcRange);
+
+    namespace refactoring {
+
+        enum TypeKind { SCALAR, VECTOR };
+
+        template<TypeKind type>
+        class GetForceRefactor : public tooling::RefactoringCallback {
+        public:
+            virtual void run(const MatchFinder::MatchResult &Result) {
+
+                const auto *call = Result.Nodes.getNodeAs<CXXMemberCallExpr>("callExpr");
+                const auto *forceName = Result.Nodes.getNodeAs<clang::StringLiteral>("forceName");
+                const Expr *objectExpr = call->getImplicitObjectArgument();
+
+                if (!call || !forceName) {
+                    std::cout << "Matched a node with no call or forceName found" << std::endl;
+                    return;
+                }
+
+                constexpr const char* kindString = (type == SCALAR) ? "scalar" : "vector"; 
```

Comment:
Would put this in an `if` statement for readability.

Also could this be a `std::string` instead of a `char *`?

Reply:
Had issues with it being constexpr and using std::string. I could make it non-constexpr to be fair.